iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RevertConnectVoltageLevelOnLine.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * This method reverses the action done in the ConnectVoltageLevelOnLine class :
+ * it replaces 2 existing lines (with the same voltage level at one of their side) with a new line,
+ * and eventually removes the voltage level in common (switching voltage level), if it contains no equipments anymore, except bus or bus bar section
+ *
+ *    VL1 ----------- switching voltage level ----------- VL2         =========>        VL1 ------------------------- VL2
+ *          (line1)                             (line2)                                              (line)
+ *
+ *
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class RevertConnectVoltageLevelOnLine extends AbstractNetworkModification {
+
+    private String line1Id;
+    private String line2Id;
+
+    private String lineId;
+    private String lineName;
+
+    /**
+     * Constructor.
+     *
+     * @param line1Id       The non-null ID of the first line
+     * @param line2Id       The non-null ID of the second line
+     * @param lineId        The non-null ID of the new line to be created
+     * @param lineName      The optional name of the new line to be created
+     */
+    RevertConnectVoltageLevelOnLine(String line1Id, String line2Id, String lineId, String lineName) {
+        this.line1Id = Objects.requireNonNull(line1Id);
+        this.line2Id = Objects.requireNonNull(line2Id);
+        this.lineId = Objects.requireNonNull(lineId);
+        this.lineName = lineName;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLine1Id(String line1Id) {
+        this.line1Id = Objects.requireNonNull(line1Id);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLine2Id(String line2Id) {
+        this.line2Id = Objects.requireNonNull(line2Id);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLineId(String lineId) {
+        this.lineId = Objects.requireNonNull(lineId);
+        return this;
+    }
+
+    public RevertConnectVoltageLevelOnLine setLineName(String lineName) {
+        this.lineName = lineName;
+        return this;
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException,
+                      ComputationManager computationManager, Reporter reporter) {
+        Line line1 = network.getLine(line1Id);
+        if (line1 == null) {
+            notFoundLineReport(reporter, line1Id);
+            if (throwException) {
+                throw new PowsyblException(String.format("Line %s is not found", line1Id));
+            } else {
+                return;
+            }
+        }
+
+        Line line2 = network.getLine(line2Id);
+        if (line2 == null) {
+            notFoundLineReport(reporter, line2Id);
+            if (throwException) {
+                throw new PowsyblException(String.format("Line %s is not found", line2Id));
+            } else {
+                return;
+            }
+        }
+
+        // Check and find the voltage level in common
```

Comment:
```
String getCommonVlIdOrThrow(Network network, boolean throwException, Reporter reporter) {
        Set<String> vlIds = new HashSet<>();
        String line1VlId1 = line1.getTerminal1().getVoltageLevel().getId();
        String line1VlId2 = line1.getTerminal2().getVoltageLevel().getId();
        String line2VlId1 = line2.getTerminal1().getVoltageLevel().getId();
        String line2VlId2 = line2.getTerminal2().getVoltageLevel().getId();
        String commonVlId = "";

        vlIds.add(line1VlId1);
        if (!vlIds.add(line1VlId2)) {
            commonVlId = line1VlId2;
        }
        if (!vlIds.add(line2VlId1)) {
            commonVlId = line2VlId1;
        }
        if (!vlIds.add(line2VlId2)) {
            commonVlId = line2VlId2;
        }

        if (vlIds.size() != 3) {
            noVoltageLevelInCommonReport(reporter, line1Id, line2Id);
            if (throwException) {
                throw new PowsyblException(String.format("Lines %s and %s should have one and only one voltage level in common at their extremities", line1Id, line2Id));
            }
            return null;
        }

     return commonVlId;
}

// [...]

String commonVlId = getCommonVlIdOrThrow(network, throwsException, reporter);
if (commonVlId == null) {
   return;
}
```

These two method may appease Sonar sensitivity to complexity.

Reply:
Code smell concerning complexity has been solved